velox/exec/tests/TableScanTest.cpp

```diff
@@ -2001,48 +1990,152 @@ TEST_F(TableScanTest, partitionedTableTimestampKey) {
   writeToFile(filePath->getPath(), vectors);
   createDuckDbTable(vectors);
   const std::string partitionValue = "2023-10-27 00:12:35";
-  testPartitionedTable(filePath->getPath(), TIMESTAMP(), partitionValue);
 
-  // Test partition filter on TIMESTAMP column.
-  {
-    auto split = exec::test::HiveConnectorSplitBuilder(filePath->getPath())
-                     .partitionKey("pkey", partitionValue)
-                     .build();
-    auto outputType =
-        ROW({"pkey", "c0", "c1"}, {TIMESTAMP(), BIGINT(), DOUBLE()});
-    ColumnHandleMap assignments = {
-        {"pkey", partitionKey("pkey", TIMESTAMP())},
-        {"c0", regularColumn("c0", BIGINT())},
-        {"c1", regularColumn("c1", DOUBLE())}};
+  auto partitionType = TIMESTAMP();
+  // Test partition value is null.
+  testPartitionedTable(filePath->getPath(), partitionType, std::nullopt);
 
-    common::SubfieldFilters filters;
-    // pkey = 2023-10-27 00:12:35.
-    auto lower = util::fromTimestampString(
-                     StringView("2023-10-27 00:12:35"),
-                     util::TimestampParseMode::kPrestoCast)
-                     .value();
-    lower.toGMT(Timestamp::defaultTimezone());
-    filters[common::Subfield("pkey")] =
-        std::make_unique<common::TimestampRange>(lower, lower, false);
+  auto split = exec::test::HiveConnectorSplitBuilder(filePath->getPath())
+                   .partitionKey("pkey", partitionValue)
+                   .build();
 
-    auto tableHandle = std::make_shared<HiveTableHandle>(
-        "test-hive", "hive_table", true, std::move(filters), nullptr, nullptr);
-    auto op = std::make_shared<TableScanNode>(
-        "0",
-        std::move(outputType),
-        std::move(tableHandle),
-        std::move(assignments));
+  ColumnHandleMap assignments = {
+      {"pkey", partitionKey("pkey", TIMESTAMP())},
+      {"c0", regularColumn("c0", BIGINT())},
+      {"c1", regularColumn("c1", DOUBLE())}};
 
+  auto sql = [&](const std::string& sqlTempl, bool asLocalTime) {
     auto t =
         util::fromTimestampString(
             StringView(partitionValue), util::TimestampParseMode::kPrestoCast)
             .thenOrThrow(folly::identity, [&](const Status& status) {
               VELOX_USER_FAIL("{}", status.message());
             });
-    t.toGMT(Timestamp::defaultTimezone());
+    if (asLocalTime) {
+      t.toGMT(Timestamp::defaultTimezone());
+    }
     std::string partitionValueStr = "'" + t.toString() + "'";
-    assertQuery(
-        op, split, fmt::format("SELECT {}, * FROM tmp", partitionValueStr));
+    return fmt::format(sqlTempl, partitionValueStr);
+  };
+
+  auto expect = [&](PlanNodePtr plan, const std::string& sqlTempl) {
```

Comment:
nit: the name 'sqlTempl' looks a bit unclear to me. Maybe use a more proper one.